gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -661,10 +690,13 @@ int doPeek() throws IOException {
       int c = nextNonWhitespace(false);
       if (c == -1) {
         return peeked = PEEKED_EOF;
-      } else {
-        checkLenient();
-        pos--;
+      } else if (!multiTopLevelValuesEnabled) {
+        throw new MalformedJsonException(
+            "Multiple top-level values support has not been enabled, use"
+                + " `JsonReader.setMultiTopLevelValuesAllowed(true)`,"
+                + locationString());
       }
+      pos--;
```

Comment:
Moved this `pos--;` outside the if-else since the `if { ... }` performs a `return`.